builder/azure/arm/config_test.go

```diff
@@ -3024,3 +3024,290 @@ func TestConfigShouldRejectSharedImageGalleryDestinationReplicationRegions(t *te
 		t.Errorf("expected config to reject with error containing %s but got %s", errorMessage, err)
 	}
 }
+
+func TestConfigShouldRejectCVMSourceToBuildManagedImage(t *testing.T) {
+	config := map[string]interface{}{
+		"image_offer":                       "ignore",
+		"image_publisher":                   "ignore",
+		"image_sku":                         "ignore",
+		"location":                          "ignore",
+		"subscription_id":                   "ignore",
+		"communicator":                      "none",
+		"managed_image_name":                "ignore",
+		"managed_image_resource_group_name": "ignore",
+		"os_type":                           constants.Target_Linux, // Does not matter for this test case, just pick one.
+		"security_type":                     "ConfidentialVM",
+		"security_encryption_type":          "VMGuestStateOnly",
+	}
+
+	var c Config
+	_, err := c.Prepare(config, getPackerConfiguration())
+	if err == nil {
+		t.Fatalf("expected config to reject with the following error: %q", "Setting a security type of \"ConfidentialVM\" is not allowed when building a VHD or creating a Managed Image, only when publishing directly to Shared Image Gallery")
+	}
```

Comment:
Can you add to these tests checks on the specific error message like the other tests in this file so this would become

```suggestion
	_, err := c.Prepare(config, getPackerConfiguration())
	errMessage := `Setting a security type of \"ConfidentialVM\" is not allowed when building a VHD or creating a Managed Image, only when publishing directly to Shared Image Gallery`
	if err == nil {
		t.Fatalf("expected config to reject with the following error: %s", errMessage)
	}
	if !strings.Contains(err.Error(), errorMessage) {
		t.Errorf("expected config to reject with error containing %s but got %s", errorMessage, err)
	}
```

Happy to help make this change too

Reply:
Ah ya this is much cleaner. Ill take you up on the help here! Feel free to commit to my branch. Appreciate the assist. 👍